Ast/src/Parser.cpp

```diff
@@ -2615,6 +2639,72 @@ AstExpr* Parser::parseString()
         return reportExprError(location, {}, "String literal contains malformed escape sequence");
 }
 
+AstExpr* Parser::parseInterpString()
+{
+    std::vector<AstArray<char>> strings;
```

Comment:
These should use TempVector to avoid excessive allocations (might need to add another std::vector for AstArray<char> to Parser)

Reply:
How would this look? I can add an `explicit TempVector(AstArray<T>& storage);`, but then I need to construct a vector for TempVector somehow. Should the `storage` field be changed from `std::vector` to something more generic?

Reply:
No, I don't think you need to change TempVector in any way - you can add `std::vector<AstArray<char>> scratchStrings;` to Parser, and then here say `TempVector<AstArray<char>> strings(scratchStrings);`. I think this should be sufficient, lmk if not.